### PR TITLE
Invites: Display invites in people management

### PR DESCRIPTION
**Pull request by jonathansadowski**
This PR adds the populated Invites tab to people management.

This section is behind a feature-flag. See p3Ex-32O-p2. Clicking on an invite row or the "Resend Invite" button is intentionally incomplete.

This PR assumes that you've applied D9507-code.

image

To test:
* Sandbox the API and apply D9507-code.
* Visit the "People" section, and then click on the "Invites" tab.
* Verify the list of invites renders appropriately.

### CODE REVIEW

File: client/my-sites/people/people-invites/index.jsx [hunk 1]
```diff
@@ -25,39 +27,62 @@ class PeopleInvites extends React.PureComponent {
 		site: PropTypes.object.isRequired,
 	};
 
-	renderInvite( invite ) {
+	renderInvite = invite => {
 		// If an invite was sent to a WP.com user, the invite object will have
 		// either a display name (if set) or the WP.com username.  Invites can
 		// also be sent to any email address, in which case the other details
 		// will not be set (but the server will still set `avatar_URL` based on
 		// the email).
```
Comment (nylen) [inline client/my-sites/people/people-invites/index.jsx]: This comment block should be moved to the appropriate place(s) in `PeopleListItem` (unless the code for each of these cases is obvious enough not to need comments).

File: client/my-sites/people/people-profile/index.jsx [hunk 2]
```diff
@@ -101,17 +106,24 @@ class PeopleProfile extends React.PureComponent {
...
 		} else if ( user.name ) {
 			name = user.name;
 		} else if ( user.label ) {
 			name = user.label;
+		} else if ( 'invite' === type && user.email ) {
+			// If an invite was sent to a WP.com user, the invite object will have
+			// either a display name (if set) or the WP.com username. Invites can
+			// also be sent to any email address, in which case the other details
+			// will not be set and we therefore display the user's email.
+			name = user.email;
```
Comment (nylen) [inline client/my-sites/people/people-profile/index.jsx]: I think this should be `user.login || user.email` (and we should remove `user.email` from the `if` condition).  Otherwise, we don't handle the case where a WP.com user has a blank display name.

File: client/my-sites/people/people-list-item/index.jsx [hunk 3]
```diff
@@ -44,40 +45,58 @@ class PeopleListItem extends React.PureComponent {
...
-					this.props,
-					'className',
-					'user',
-					'site',
-					'isSelectable',
-					'onRemove',
-					'moment',
-					'numberFormat',
-					'translate'
-				) }
```
Comment (nylen) [inline client/my-sites/people/people-list-item/index.jsx]: Nice job getting rid of this.

File: client/my-sites/people/people-list-item/index.jsx [hunk 4]
```diff
@@ -44,40 +45,58 @@ class PeopleListItem extends React.PureComponent {
...
+		const { invite, site, type, user } = this.props;
+		const editLink = this.canLinkToProfile() && `/people/edit/${ site.slug }/${ user.login }`;
+		const inviteLink = invite && `/people/invites/${ site.slug }/${ invite.invite_key }`;
+
+		return type === 'invite' ? inviteLink : editLink;
+	};
+
+	renderInviteStatus = () => {
+		const { invite, translate } = this.props;
+		const { is_pending: isPending } = invite;
```
Comment (nylen) [inline client/my-sites/people/people-list-item/index.jsx]: We can get rid of this awkwardness by transforming the invites objects we receive from the API as they are inserted into Redux state.  This is a good idea for other reasons too:  https://github.com/Automattic/wp-calypso/pull/21505#discussion_r161935011

I don't have a strong preference whether this should be done here or in a separate PR.  Probably a separate PR so that we can keep moving.

File: client/my-sites/people/people-list-item/index.jsx (hunk 3 above)
Comment (jonathansadowski) [reply client/my-sites/people/people-list-item/index.jsx]: Yeah -- it wasn't used anywhere and seemed like it'd be hard to maintain. I think being explicit about props is a better practice.

**Comment by nylen** [comment]
There is a `follower` role for invites which needs to be handled [in `PeopleProfile`](https://github.com/Automattic/wp-calypso/blob/b4ed427b/client/my-sites/people/people-profile/index.jsx#L35-L75):

2018-01-24t20 24 59-0600

I noticed this immediately after approving the PR.  Feel free to merge after fixing that.

**Comment by jonathansadowski** [comment]
Thanks for the review -- I handled the follower role before merging, as it was a small change. If we want to treat the style of the badge differently, we can do that in a future PR, but at least this will make sure the string is wrapped for translation.
